### PR TITLE
bitmask-vpn: install the app's icon

### PR DESCRIPTION
**Pull request by hyphenrf**
###### Description of changes

Just an aesthetics thing; currently the apps are installed onto graphical interfaces with the default icon.
This does not change the build process, merely installs an additional file during postInstall.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/bitmask-vpn/default.nix [hunk 1]
```diff
@@ -78,6 +78,9 @@ buildGoModule rec {
     substituteInPlace providers/vendor.conf \
       --replace "provider = riseup" "provider = ${provider}"
 
+    substituteInPlace $out/share/applications/${provider}-vpn.desktop \
```
Comment (ju1m) [inline pkgs/tools/networking/bitmask-vpn/default.nix:81]: `$out` is not created yet when in `postPatch`, so this does not work, instead try to substitute in `branding/templates/debian/app.desktop-template`

File: pkgs/tools/networking/bitmask-vpn/default.nix [hunk 2]
```diff
@@ -78,6 +78,9 @@ buildGoModule rec {
...
+      --replace icon ${provider}-vpn
```
Comment (ju1m) [inline pkgs/tools/networking/bitmask-vpn/default.nix:82]: `--replace Icon=icon Icon=${provider}-vpn` would be clearer, but that's me nitpicking.

**Review by ju1m — CHANGES_REQUESTED** [review]
Hi @hyphenrf, thanks for submitting this change. Could you please fix the `substituteInPlace` and tell us where do you expect this icon to appear? In a launching menu?

File: pkgs/tools/networking/bitmask-vpn/default.nix (hunk 1 above)
Comment (hyphenrf) [reply pkgs/tools/networking/bitmask-vpn/default.nix:81]: Ah yes my bad, I moved it from `postInstall` and forgot to fix the path. Sorry for the silly mistake and thanks for catching it!

File: pkgs/tools/networking/bitmask-vpn/default.nix (hunk 2 above)
Comment (hyphenrf) [reply pkgs/tools/networking/bitmask-vpn/default.nix]: I agree it's more intention-signalling, done!

**Comment by hyphenrf** [comment]
> tell us where do you expect this icon to appear? In a launching menu?

yeah exactly, like this:
image
instead of gnome's default app icon

**Comment by hyphenrf** [comment]
@ju1m please let me know if you'd like me to squash the changes

**Comment by ju1m** [comment]
Result of `nixpkgs-review pr 168904` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>calyx-vpn</li>
    <li>riseup-vpn</li>
  </ul>
</details>

**Comment by ju1m** [comment]
@hyphenrf the change is small so it's not a big deal, the only advice I can give you is that it's common to force push commits in Nixpkgs's PRs to minimize the cognitive overhead for new reviewers and be more accommodating to `git bisect`.

**Review by ju1m — APPROVED** [review]

**Comment by hyphenrf** [comment]
I believe the option exists for maintainers to merge with a squash, but if it's preferable to do that on my side, I'll do it right away.

**Pull request merged**